### DISCUSSION TITLE
[Prisma 4] Connection hanging indefinitely on queries and operations under preassure

### DISCUSSION THREAD

**Discussion by alesk20** [prisma/prisma#24039] [Q&A] [answered]
### Bug description

Hello, I have nodejs server that has a lot of traffic and has a lot of continuing write/read operation on the database.
I am using Prisma 4 version with a Postgres database: I'm actually using @prisma/client 4.10.1 and having no problems. 

Upgrading to the last released Prisma 4 version instead I have a problem: after 1-2 hours the connection to the database start to hang indefinitely to new queries and operations. I don't get any timeout error, the Promise just hang indefinitely until I have the necessity to restart the server because with time always more queries/operations begin to hang indefinitely blocking my server internal operations.

My question is: is there any known issue after Prisma 4.10.1 version that provides this type of problem?

**Comment by ludralph** [comment, marked as answer]
Hi @alesk20 👋 

Thank you for raising this question.
 
There have been no reported performance issues specifically associated with Prisma Version 4.10.1. In fact, a [previously reported issue](https://github.com/prisma/prisma/issues/19094) concerning performance was later determined to be unrelated to Prisma.

It's important to note that significant performance enhancements have been incorporated into recent releases of Prisma, starting from [Version 5](https://github.com/prisma/prisma/releases/tag/5.0.0). We strongly recommend updating to the latest [release](https://github.com/prisma/prisma/releases) to take advantage of these improvements.

**Comment by alesk20** [comment]
Hi @ludralph,
as you said, I actually find out the problem and it wasn't related to Prisma. I'll eventually update to version 5 to get more improvements and better performances, but for I have no problem with version 4 and I can use it as a stable version
Thank you for the help.

Kind regards.
